tools/fuzzer.cpp

```diff
@@ -0,0 +1,274 @@
+////////////////////////////////////////////////////////////////////////////////
+/// @brief Library to build up VPack documents.
+///
+/// DISCLAIMER
+///
+/// Copyright 2022 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Julia Puget
+/// @author Copyright 2022, ArangoDB GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+#include <bitset>
+#include <charconv>
+#include <iostream>
+#include <random>
+#include <string>
+#include <thread>
+
+#include "velocypack/vpack.h"
+#include "velocypack/velocypack-exception-macros.h"
+
+using namespace arangodb::velocypack;
+
+enum class Format {
+  VPACK, JSON
+};
+
+enum RandomBuilderAdditions {
+  ADD_ARRAY = 0,
+  ADD_OBJECT,
+  ADD_BOOLEAN,
+  ADD_STRING,
+  ADD_NULL,
+  ADD_UINT64,
+  ADD_INT64,
+  ADD_DOUBLE, // here starts values that are only for vpack
+  ADD_UTC_DATE,
+  ADD_BINARY,
+  ADD_EXTERNAL,
+  ADD_ILLEGAL,
+  ADD_MIN_KEY,
+  ADD_MAX_KEY,
+  ADD_MAX_VPACK_VALUE
+};
+
+struct RandomGenerator {
+  RandomGenerator() : mt64{rd()} {}
+
+  std::random_device rd;
+  std::mt19937_64 mt64;
+};
+
+std::vector<std::jthread> jthreads;
```

Comment:
I don't think `std::jthread` is available on all platforms yet. According to https://en.cppreference.com/w/cpp/compiler_support, not even clang supports it. Please use `std::thread`.